packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -206,6 +185,39 @@ export class Arn {
     });
   }
 
+  /**
+   * Extract the full resource name from an ARN
+   *
+   * Necessary for resource names (paths) that may contain the separator, like
+   * `arn:aws:iam::111111111111:role/path/to/role/name`.
```

Comment:
Can you clarify when to use `parse()` vs `extractResourceName()` here, and what's recommended in most cases?